src/coreComponents/physicsSolvers/multiphysics/CoupledSolver.hpp

```diff
@@ -144,9 +144,23 @@ class CoupledSolver : public SolverBase
                      real64 const & dt,
                      DomainPartition & domain ) override
   {
+    Timestamp const meshModificationTimestamp = getMeshModificationTimestamp( domain );
+
     forEachArgInTuple( m_solvers, [&]( auto & solver, auto )
     {
+      // Only build the sparsity pattern if the mesh has changed
+      if( meshModificationTimestamp > solver->getSystemSetupTimestamp() )
```

Comment:
isn't this done in the individual solvers?

Reply:
probably yes but this is what sequential was doing, I just moved it into a function
let me double check

Reply:
seems that timestamp logic is outside of `implicitStepSetup` for some reason, see
https://github.com/GEOS-DEV/GEOS/blob/decc28110acf58053a0d6911f5ea7a4d9d7f76d1/src/coreComponents/physicsSolvers/SolverBase.cpp#L215

Reply:
@CusiniM i had to revert this change - there is some logic mismatch for hydrofrac solver that leads to
```
Time: 0.00e+00 s, dt: 1 s, Cycle: 0


***** LOCATION: /data/rpo_ptls/GEOSX/residual_flash/GEOS/GEOS/src/coreComponents/dataRepository/Group.hpp:1118
***** Controlling expression (should be false): wrapper == nullptr
***** Rank 0: Group /domain/MeshBodies/mesh1/meshLevels/Level0/ElementRegions/elementRegionsGroup/Fracture(kgdBase_C3D6_base.xml,l.24)/elementSubRegions/FractureSubRegion has no wrapper named FE1
The wrappers of FractureSubRegion are: { localToGlobalMap, globalToLocalMap, isExternal, ghostRank, domainBoundaryIndicator, numNodesPerElement, numEdgesPerElement, numFacesPerElement, elementCenter, elementVolume, nodeList, edgeList, fractureElementsToCellRegions, fractureElementsToCellSubRegions, fractureElementsToCellIndices, elementAperture, elementArea, creationMass, dNdX, detJ, faceList, edgesToFractureConnectors, fractureConnectorsToEdges, fractureConnectorsToElementIndex, elem2dToCollocatedNodesBuckets, water_density, water_dDensity_dPressure, water_dDensity_dTemperature, water_density_n, water_viscosity, water_dViscosity_dPressure, water_dViscosity_dTemperature, water_internalEnergy, water_internalEnergy_n, water_dInternalEnergy_dPressure, water_dInternalEnergy_dTemperature, water_enthalpy, water_dEnthalpy_dPressure, water_dEnthalpy_dTemperature, rock_stress, rock_oldStress, rock_density, rock_thermalExpansionCoefficient, rock_bulkModulus, rock_shearModulus, fracturePorosity_porosity, fracturePorosity_porosity_n, fracturePorosity_dPorosity_dPressure, fracturePorosity_dPorosity_dTemperature, fracturePorosity_initialPorosity, fracturePorosity_referencePorosity, fracturePerm_permeability, fracturePerm_dPerm_dPressure, fracturePerm_dPerm_dDispJump, porousMaterialNames, porosityModelName, solidMaterialNames, fluidNames, solidNames, permeabilityNames, gravityCoefficient, netToGross, pressure, pressure_n, initialPressure, deltaPressure, bcPressure, temperature, temperature_n, initialTemperature, bcTemperature, aperture_n, hydraulicAperture, minimumHydraulicAperture, deltaVolume, mobility, dMobility_dPressure, K_IC_00, K_IC_01, K_IC_02, K_IC_10, K_IC_11, K_IC_12, K_IC_20, K_IC_21, K_IC_22, ruptureTime, ruptureRate, SinglePhaseFlow_primaryVariables_dofIndex }
```